docs/DeveloperGuide.md

```diff
@@ -4,11 +4,212 @@
+
+```
+THIS IS AN IRREVERSIBLE PROCESS. ARE YOU SURE? Y/N
+```
+
+**Step 3.1.** The user decides against clearing the ```TaskList```, and inputs a ```N```. The ```TaskList``` has no changes. ```Duck``` resumes after the output in the following format has been displayed:
+
+```
+
+Process cancelled.
+
+```
+
+**Step 3.2.** The user decides upon clearing the ```TaskList```, and inputs a ```Y```. The ```TaskList``` has been cleared.
+
+![](https://lh4.googleusercontent.com/G4PalSoaVQxExSGKCDefJuO4TlyCJwrCfZDT4BsTrrSHMJ8aQSPsaxGoFJYb9YIO1yo3_3nV0Jar3Haqfac90v7G_yfZdx8-OwxSKEART5zLqMZ73k3YW0ssPbIE3kOhkbeusR3jVMdD6kTSkG7rK4s)
+
+```Duck``` resumes after the output in the following format has been displayed:
+
+```
+
+Got it, all tasks have been cleared.
+
+ ```
+
+### SchoolClass Feature
+
+**Implementation**
+
+The ```SchoolClass``` Class is implemented to facilitate the adding of students' classes to the task list. It extends the ```Task``` Class with additional String attributes to store the class name, start date/time, and end date/time. It also overrides the ```toString()``` method to have its own specialised output when being printed, as well as overriding the ```toSaveString()``` method to have correctly save its details to the save file. The ```SchoolClass``` Class will also facilitate the implementation of automatically recurring classes, which is a planned feature for milestone v2.0. ```SchoolClass``` Tasks will be added by default as recurring tasks to the task list, and will be automatically added back at their same set timing each week.
+
+The following are the new operations implemented.
+
+-   ```TaskList#addSchoolClass()``` - Adds a SchoolClass Task to the task list.
+
+-   ```Ui#addedTaskMessage()``` - Outputs a message to show that the SchoolClass Task was successfully added to the task list
+
+-   ```Storage#loadSchoolClass()``` - Adds a SchoolClass to the task list without generating a successfully added message, to be used when loading from the save data.
+
+<br />
+
+**Given below is an example usage scenario for  TaskList#addSchoolClass().**
+
+**Step 1.** The user inputs a command following the proper formatting for adding a ```SchoolClass```. The Duck class will call ```Parser#processCommand()```, instantiating a ```Parser``` class, which will then call ```TaskList#addTask()```, instantiating a ```TaskList``` Class. From there, ```TaskList#addSchoolClass()``` is called and a new ```SchoolClass``` Task is instantiated, which calls ```Ui#addedTaskMessage()``` and instantiating a ```Ui``` Class. This ```SchoolClass``` Task will thus be added to the ArrayList<Task> tasks that was instantiated in the ```Duck``` class.
+
+![](https://lh4.googleusercontent.com/u4zVr8TYxFw3rMvnqdwCYlJmq0JxUgEtC_cFmOY7rPqCM9nvzcQL1t-GcTmgbedeVEHi2L6MG6xG3QaJ7XaOPs8nYvHz1Uf4wGK9bMsHDHwxZVNdS2zR79TtHL_Ub2Za0_jm6bUsnY_RQWX6QmFqCl4)
```

Comment:
Since this example scenario involves quite a few method calls between objects, wouldn't a sequence diagram be more appropriate? An object diagram only shows how the objects are connected together but does not show functionality.
<img width="982" alt="Screen Shot 2023-03-30 at 15 37 39" src="https://user-images.githubusercontent.com/88705750/228764024-8a7b3fda-fd87-459d-8afb-5eea204ce256.png">